usersession/userd/launcher.go

```diff
@@ -58,6 +64,7 @@ const launcherIntrospectionXML = `
 
 var (
 	allowedURLSchemes = []string{"http", "https", "mailto", "snap", "help"}
+	allowedEnvVars    = []string{"DISPLAY", "WAYLAND_DISPLAY", "XDG_CURRENT_DESKTOP", "XDG_SESSION_DESKTOP", "XDG_SESSION_TYPE"}
```

Comment:
This is possibly a reason to make the interface "superprivileged" since a snap that can modify these environment variables might be able to DoS the app, make the app connect to somewhere else or make it behave differently from the available session. This is not a blocker since this interface obviously needs to set these for the app; it just might be a reason for an installation constraint.

Reply:
Is "superprivileged" documented anywhere? I'm not sure what the implications are.

Reply:
I think the difference Jamie is referring to is between an interface being disconnected by default (but the user can hook up with `snap connect`), and not even being able to run `snap connect` without an assertion from the store specifically allowing it.

Reply:
For _my_ principle use case (a confined shell) a store assertion would be acceptable (how many confined shells are there likely to be?). But I can imagine additional applications wanting to use this facility and that may be less clear-cut.